src/core/tokenizer.ts

```diff
@@ -91,12 +91,13 @@ function generateMatcher(
   };
 }
 
-const matchers = new Map<string[], Matcher>();
+const matcherCache: Record<string, Matcher> = {};
 
 export function getMatcher(languageId: string): Matcher {
   const wordSeparators = tokenizerConfiguration.getWordSeparators(languageId);
+  const cacheKey = JSON.stringify(wordSeparators);
```

Comment:
I don't think this is an improvement at all. You're just wasting resources with something that maps are super optimized to do. The amount of times this function gets called in in a single second is quite high.

Reply:
See my comment; maps compare keys by equality

Reply:
Feel free to switch it back to a map, but you need to stringify.  See eg https://stackoverflow.com/a/69219091

Reply:
I know